### DISCUSSION TITLE
Non-English Languages in TextAdept

### DISCUSSION THREAD

**Discussion by iloveyou3869** [orbitalquark/textadept#382] [General]
I have tried a foreign language keyboard in TextAdept and the one I tried didn't function, TextAdept ignored my keyboard choice and only received English as its input.

For composing non-programming related documents I often use three or more languages.
Can I use more than one language in a document?
Does TextAdept support using Right-to-Left writing?

**Comment by orbitalquark** [comment]
Keybindings for foreign keyboards will not work out of the box: 
https://github.com/orbitalquark/textadept/issues/64. Typing foreign 
language characters should work though.

By default, buffers contain UTF-8 text, so Textadept can have multiple 
language characters in a single document, provided your font supports 
the correct glyphs.

Textadept does not support RTL writing.
